src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -621,12 +621,8 @@ private SubpatternSyntax ParseSubpatternElement()
         /// </summary>
         private bool IsPossibleSubpatternElement()
         {
-            var tk = this.CurrentToken.Kind;
-            bool isExpression = this.IsPossibleExpression() &&
-                    // IsPossibleExpression returns true when the next token is a binary operator.
-                    // That is useful for error recovery elsewhere, but not here.
-                    !(SyntaxFacts.IsBinaryExpression(tk) || SyntaxFacts.IsAssignmentExpressionOperatorToken(tk));
-            return isExpression || this.CurrentToken.Kind == SyntaxKind.OpenBraceToken;
+            return this.IsPossibleExpression(allowBinaryExpressions: false, allowAssignmentExpressions: false) ||
+                this.CurrentToken.Kind == SyntaxKind.OpenBraceToken;
```

Comment:
Thanks!  Note: i would personally find it valuable if there was an explanation here (even if brief) as to why neither of these expression forms are allowed.  i.e. what hte intuition here is for why this is correct.

Reply:
These parameters are about whether binary operator tokens should be accepted.  We don't want a token such as `==` to be considered to be permitted in this context.